src/audiomixcontroller/DefaultAudioMixController.ts

```diff
@@ -56,13 +55,13 @@ export default class DefaultAudioMixController implements AudioMixController {
         const oldSinkId: string = this.audioElement.sinkId;
         if (newSinkId !== oldSinkId) {
           if (this.browserBehavior.hasChromiumWebRTC()) {
-            new AsyncScheduler().start(async () => {
-              const existingStream = await this.audioElement.srcObject;
-              this.audioElement.srcObject = null;
-              // @ts-ignore
-              await this.audioElement.setSinkId(newSinkId);
-              this.audioElement.srcObject = existingStream;
-              this.audioStream = existingStream as MediaStream;
+            const existingStream = this.audioElement.srcObject;
```

Comment:
What will happen if the user changes mind and calls it with another element?
```
bindAudioStream(streamA);
bindAudioElement(elementA);
// changed mind
bindAudioStream(streamB);
bindAudioElement(elementB);

// elementA.srcObject will reference streamB!
```

Instead of the stream, how about saving the element itself temporarily? In the callback, you can do the following:
```
const existingAudioElement = this.audioElement;
this.audioElement.srcObject = null;
// @ts-ignore
this.audioElement.setSinkId(newSinkId).then(() => {
  if (this.audioElement === existingAudioElement) {
     this.audioElement.srcObject = this.audioStream;
  }
});
```

Reply:
Great Catch